lib/internal/main/watch_mode.js

```diff
@@ -64,11 +66,23 @@ let exited;
 
 function start() {
   exited = false;
+
+  let kEnv = {};
+  for (const arg of process.execArgv) {
+    if (StringPrototypeStartsWith(arg, '--env-file')) {
+      const envFile = resolve(StringPrototypeSplit(arg, '=')[1]);
+      watcher.filterFile(envFile, null);
+      const envContent = parseEnv(readFileSync(envFile, 'utf8'));
+      kEnv = { ...kEnv, ...envContent };
```

Comment:
```suggestion
```
I am pretty sure that `watcher.filterFile` is enough since the spawned process should handle `--env-file` by its own
this is also missing error handling (what happens if env file is renamed/deleted)?

Reply:
> I am pretty sure that `watcher.filterFile` is enough since the spawned process should handle `--env-file` by its own

I've tested it many times, but when the child is spwan, it inherits the parent's `process.env`. If I only use the `watcher.filterFile`, if the env file is modified, it triggers a restart of the child process, but the information from the modified env doesn't seem to go to the child.

This is why I used `parseEnv` and `readFileSync`.

Refs: https://github.com/nodejs/node/issues/54001#issuecomment-2245219748

> this is also missing error handling (what happens if env file is renamed/deleted)?

Oh, this is my mistake.

If I can't find the --env-file I set when I started watch mode (because it's renamed or deleted), is it a good way to print `${envFile}: not found` like when I first start watch mode?